src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/combat/ModuleAutoRod.kt

```diff
@@ -0,0 +1,193 @@
+                        }
+                    }
+                }
+
+                // Check whether player is using items/blocking.
+                if (!onUsingItem) {
+                    if (mc.player?.mainHandStack != Items.FISHING_ROD &&
+                        (mc.player?.isUsingItem == true) ||
+                        KillAuraAutoBlock.blockVisual
+                    ) {
+                        return@tickHandler
+                    }
+                }
+
+                if (facingEntity?.isAttackable == true) {
+                    // Checks how many enemy is nearby, if <= then should rod.
+                    var enemieslist =
+                        targetTracker.enemies().filter { it.isAttackable && player.distanceTo(it) < range }
+                    enemieslist.size.let { it1 ->
+                        if (it1 <= enemiesNearby) {
+                            // Check if the enemy's health is below the threshold.
+                            if (ignoreOnEnemyLowHealth.enabled) {
+                                if ((facingEntity is LivingEntity) &&
+                                    facingEntity.getActualHealth() >= ignoreOnEnemyLowHealth.enemyHealthThreshold
+                                ) {
+                                    rod = true
+                                }
+                            } else {
+                                rod = true
+                            }
+
+                        }
+                    }
+                } else mc.player?.health?.let { it1 ->
+                    if (it1 <= escapeHealthThreshold) {
+                        // use rod for escaping when health is low.
+                        rod = true
+                    } else if (!facingEnemy.enabled) {
+                        // Rod anyway, spam it.
+                        rod = true
+                    }
+                }
+            }
+
+            if (rod && pushTimer.hasElapsed(pushDelay.toLong())) {
+                // Check if player has rod in hand
+                if (mc.player?.mainHandStack != Items.FISHING_ROD) {
+                    // Check if player has rod in hotbar
+                    val rod = Slots.Hotbar.findSlot(Items.FISHING_ROD)?.hotbarSlot
+                    if (rod == null) {
+                        // There is no rod in hotbar
+                        return@tickHandler
+                    }
+                    // Switch to rod
+                    switchBack = mc.player?.inventory?.getSlotWithStack(mc.player?.inventory?.mainHandStack)
+                    mc.player?.inventory?.selectedSlot = rod
+                    interaction.syncSelectedSlot()
+                }
+                rod()
+            }
+        }
+    }
+
+    /**
+     * Use rod
+     */
+    private fun rod() {
+        val rod = Slots.Hotbar.findSlot(Items.FISHING_ROD)?.hotbarSlot
+        if (rod == null) {
+            return
+        }
+        mc.player?.inventory?.selectedSlot = rod
```

Comment:
We  do not change the selected slot like that in LiquidBounce Nextgen. LiquidBounce Nextgen offers a much util for that instead called `SilentHotbar`.